### PR TITLE
Fix combo box can be focused when disabled

### LINKED ISSUE #3369: Disabled combo box stays focused
### Package name(s)

Core (@salt-ds/core)

### Package version(s)

"@salt-ds/core": "^1.15.0",

### Description

For our use case the combobox is disabled after the user selects an option. The issue is that the combo box stays in focus (shown by the dotted blue lines around the combobox). Even when a user clicks on something else the diabled combobox stays focused.

### Steps to reproduce

set disabled set to true on onSelectionChange if a option is selected 

### Expected behavior

Once the combo box disables and the user click on another part of the screen then the combobox should lose focus. 

### Operating system

- [ ] macOS
- [X] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Browser

- [X] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

### Are you a JPMorgan Chase & Co. employee?

- [X] I am an employee of JPMorgan Chase & Co.

### PR DESCRIPTION
**Pull request by origami-z**
Try fix #3369

### CODE REVIEW

File: packages/core/src/combo-box/ComboBox.tsx [hunk 1]
```diff
@@ -134,6 +134,12 @@ export const ComboBox = forwardRef(function ComboBox<Item>(
     removePill,
   } = listControl;
 
+  useEffect(() => {
```
Comment (origami-z) [inline packages/core/src/combo-box/ComboBox.tsx:137]: @joshwooding  i don't quite like this, but can't think of another way

**Comment by joshwooding** [comment]
Hmm, we might need to check disabled onFocus again

**Review by joshwooding — APPROVED** [review]

File: packages/core/src/__tests__/__e2e__/combo-box/ComboBox.cy.tsx [hunk 2]
```diff
@@ -306,6 +308,38 @@ describe("Given a ComboBox", () => {
     cy.findByRole("button", { name: "end" }).should("be.focused");
   });
 
+  it("should not receive focus via mouse click if it is disabled", () => {
+    cy.mount(<Disabled />);
+    cy.findByRole("combobox").realClick();
+    cy.get(".saltComboBox").should("not.have.class", "saltComboBox-focused");
```
Comment (joshwooding) [inline packages/core/src/__tests__/__e2e__/combo-box/ComboBox.cy.tsx:314]: Should we test actual focus too? e,g, be.focused

File: packages/core/src/__tests__/__e2e__/dropdown/Dropdown.cy.tsx [hunk 3]
```diff
@@ -240,6 +241,36 @@ describe("Given a Dropdown", () => {
     cy.findByRole("button", { name: "end" }).should("be.focused");
   });
 
+  it("should not receive focus via mouse click if it is disabled", () => {
+    cy.mount(<Disabled />);
+    cy.findByRole("combobox").realClick();
+
+    cy.findByRole("combobox").should("be.disabled");
```
Comment (joshwooding) [inline packages/core/src/__tests__/__e2e__/dropdown/Dropdown.cy.tsx:248]: This technically doesn't test focus only if it's disabled and the list doesn't open

**Review by joshwooding — APPROVED** [review]

**Review by Fercas123 — APPROVED** [review]

**Pull request merged**